2.35.0 release notes (#977)
percona/pmm-doc: docs/release-notes/2.35.0.md

---
name: Percona Monitoring and Management <version>
about: Release Notes Template for PMM

---

| **Release date:** | [Month DD, YYYY]                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

!!! caution alert alert-warning "Important/Caution"
    Crucial points that need emphasis:

    - Important: A significant point that deserves emphasis.
    - Caution: Used to mean 'Continue with care'.

## Release Highlights

###  Access Control: Limit access to metrics 
!!! caution alert alert-warning "Important/Caution"
    Disclaimer:  PMM Access Control is still technical preview and is subject to change. We recommend that early adopters use this feature for testing purposes only.

> We are excited to announce the first step towards PMM Access Management, a highly anticipated solution by our users. PMM now allows you to limit access to metrics, based on labels. This new feature provides a standardized method for granting, changing, and revoking access to metrics. You can now establish policies that control which data can be queried based on specific label criteria with the introduction of label-based access control. It allows you to associate multiple sets of individual Prometheus (Victoria Metics) labels with a policy, ensuring that queries will only return data from series that meet at least one of the defined labels. This provides you with a powerful and flexible way to manage your data and ensure secure access to the information you need. For more information, please visit the Access Control page. 

Reviewer: "Prometheus (Victoria Metics) labels"
We should avoid using technical language in Highlights, as this supports "oh, it's Prom + Grafana ...."

Author: ok

To ensure a seamless rollout, PMM Access Management will be introduced in phases, taking into consideration your requirements and feedback. Stay tuned for further updates on our roadmap for PMM Access Management..

###  Deploying PMM in k8 with Helm Chart is now generally available
We are thrilled to announce the general availability of deploying PMM in Kubernetes with Helm chart. The deployment of PMM in Kubernetes via Helm chart has been available in Tech Preview since the release of PMM v2.29, and we are now proud to offer it to our users as a supported feature in general availability. For more information, see the  [documentation](https://docs.percona.com/percona-monitoring-and-management/setting-up/server/helm.html).  

###  Ability to restore cluster from backup stored on S3 while creating DB clusters in DBaaS

> ### Ability to restore cluster from backup stored on S3 while creating DB clusters in DBaaS

Reviewer: ```suggestion
```
duplication of the line ^

Starting with PMM 2.35.0, we are excited to announce the first step towards PMM Access Management, a highly anticipated solution by our users. With this solution, PMM allows you to manage who has access to individual Prometheus (Victoria Metics)  metrics based on labels. This new feature provides a standardized method for granting, changing, and revoking access to metrics. With the introduction of label-based access control, you can now establish roles that control which data can be queried based on specific label criteria. It allows you to associate multiple sets of individual Prometheus (Victoria Metics) labels with a role, ensuring that queries will only return data from series that meet the defined labels. This provides you with a powerful and flexible way to manage your data and ensure secure access to the information you need. For more information, visit the Access Control documentation.

PMM Access Management will be introduced in phases to ensure a seamless rollout in line with your requirements and feedback. Stay tuned for further updates on our roadmap for PMM Access Control.

> ###  Deploying PMM in k8 with Helm Chart is now generally available

Reviewer 2: ```suggestion
###  Deploying PMM in Kubernetes with Helm Chart is now Generally Available
```

# Percona Monitoring and Management 2.35.0

| **Release date:** | [Feb 21, 2023]                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open-source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

## Release Highlights

### Access Control: Limit access to metrics

!!! caution alert alert-warning "Important/Caution"
    Disclaimer:  PMM Access Control is still in Technical Preview and is subject to change. We recommend that early adopters use this feature for testing purposes only.

> Starting with PMM 2.35.0, we are excited to announce the first step towards PMM Access Management, a highly anticipated solution by our users. With this solution, PMM allows you to manage who has access to metrics based on labels. This new feature provides a standardized method for granting, changing, and revoking access to metrics. With the introduction of label-based access control, you can now establish roles that control which data can be queried based on specific label criteria. It allows you to associate multiple sets of labels with a role, ensuring that queries will only return data from series that meet the defined labels. This provides you with a powerful and flexible way to manage your data and ensure secure access to the information you need. For more information, visit the Access Control documentation.

Reviewer 3: ```suggestion
We're thrilled to announce that with the release of PMM 2.35.0, we've taken the first step towards introducing PMM Access Control, a highly requested solution by our users. Label-based access control in PMM allows you to manage who has access to metrics based on labels. You can now create roles that determine which data can be queried based on specific label criteria, such as allowing the QA team to view test environment data on dashboards. By associating multiple sets of labels with a role, this feature ensures that only data from series that meet the defined labels will be returned. This new feature provides a standardized method for granting, changing, and revoking access to metrics, giving you a powerful and flexible way to manage access to metrics and ensure secure access to the information you need. For further information, please refer to the Access Control documentation.
```

Author: Modified

> 

Author: Yes. I will

#### Cluster membership check

To avoid execution errors, PMM no longer allows scheduling backups on MongoDB Services that are not managed as a cluster.

You can check which of your current Services are not members of a cluster on the **PMM Inventory > Services** page. Services that do not specify a cluster name in the **Other Details** column should be removed and re-added using the following command:

<pre>pmm-admin add mongodb
username=pmm_mongodb --password=password 
query-source=profiler <mark>--cluster=mycluster</mark></code></pre>

### Backup restore logs accessible from PMM

With this release, we are adding the option to check the logs for restores from a MongoDB database backup.
This complements the existing option to check the logs for the backup process and makes it easier to gather data for compliance, auditing, and troubleshooting purposes.

Instead of having to look through PBM log files, you can now access the **Log** option in PMM, which is now automatically displayed next to each MongoDB backup restore entry on the **Backup > Restores** page.

### Components Upgrade

> MongoDB exporter has been upgraded to 0.37.0.

Reviewer 3: @rasika-chivate can we also update this page with new exporter version? https://docs.percona.com/percona-monitoring-and-management/details/pmm_components_and_versions.html

Author: Yes. it will be done before the Release..

Author: Done.

## New Features

> - [PMM-10325](https://jira.percona.com/browse/PMM-10325) - Starting with PMM 2.35.0, for enhanced security, you can change the admin password even after `pmm-agent` is registered on the PMM Server.

Reviewer 3: @rishat-ishbulatov @rasika-chivate this explanation doesnt give the story :) Could we add the problem and how we solved it in brief?

Author: Is the user really interested in how we solved the problem?

Author: Resolving this.

- [PMM-11357](https://jira.percona.com/browse/PMM-11357) - [Percona Alerting]: Fixed an issue where the name of an alert rule was not inherited correctly from the alert template.
- [PMM-11399](https://jira.percona.com/browse/PMM-11399) - [Percona Alerting]: The pop-up message that confirmed saving an alert rule would not display the name of the rule correctly. This issue is now resolved.
- [PMM-11517](https://jira.percona.com/browse/PMM-11517) - [Backup Management]: Fixed a PBM synchronization issue that made PMM unable to restore local physical backups after the first attempt.

<!--- ## Known issues

- ​List of known issues with a  comprehensive description and link to the JIRA ticket.

    Example:

    [PMM-XXXX](https://jira.percona.com/browse/PMM-XXXX) - Comprehensive description.

    **Solution**

    Description of the solution. --->

## Coming Soon

### Access Management

The following features/use cases will be covered as part of the various stages of PMM Access Management:

> Role-based access control, including access limitations to the features like **Alerting** and **Backup** with read-edit-delete permissions.

Reviewer 3: ```suggestion

> Role-based access control (RBAC), including access limitations to the features like **Alerting** and **Backup** with read-edit-delete permissions.

> PMM High Availability (HA) is coming soon! The [planned phases](https://docs.percona.com/percona-monitoring-and-management/release-notes/2.33.0.html#offline-metric-collection) will be launched soon. 

```

### GA: Deployment of PMM in Kubernetes with Helm Chart

We are happy to announce the General Availability of PMM deployment in Kubernetes with the Helm chart. PMM deployment via Helm chart in Kubernetes has been available as Tech Preview since PMM v2.29, but now we're delighted to offer it to users as a fully supported feature. For more information, see [documentation](https://docs.percona.com/percona-monitoring-and-management/setting-up/server/helm.html).  

### DBaaS: Backup schedule while creating DB clusters

Starting with PMM 2.35.0, you can add a backup schedule while creating DB clusters in DBaaS. This feature is a fusion of Backup management and DBaaS in PMM. Currently, DBaaS only supports scheduled backups, which can only be enabled when a database cluster is created.

!!! note alert alert-primary "Note"
    DBaaS does not support on-demand backups.

> ### BaaS: Restore cluster from backup stored on S3

Reviewer: ```suggestion
### DBaaS: Restore cluster from backup stored on S3
```

> You can now create a DBaaS cluster from a backup stored on S3. Currently, you cannot restore a cluster from a backup. You can only create a DBaaS database cluster from backup.

Reviewer 4: Sharing some feedback from @nethalo (slightly rephrased):
```suggestion
You can now create a DBaaS cluster from a backup stored on S3. Backups from an existing cluster can only be used to create a NEW DBaaS cluster. Attempts to use it to restore a previous version of the cluster will result in downtime since the system will treat it as a creation of a new cluster.
`

- [11302](https://jira.percona.com/browse/PMM-11302) - [DBaaS]: Starting with PMM 2.35.0, you can add a backup schedule while creating DB clusters in DBaaS.

- [PMM-10962](https://jira.percona.com/browse/PMM-10962) - Starting with PMM 2.35.0, we have introduced Access Control. Roles are a vital part of Access Control that provide the users with access to specified metrics. You can create, edit, and delete user roles. The roles can be assigned to the users with the **Configuration > Users** tab. For more information, see [documentation]().

## Improvements

- [PMM-10471](https://jira.percona.com/browse/PMM-10471) - [Backup Management]: PMM now automatically checks the current MongoDB node to ensure that the connected PMM Agents have the tools to execute and finalize the backup.

- [PMM-11113](https://jira.percona.com/browse/PMM-11113) - Enhanced user experience for the **Percona Platform** page under **Configuration > Settings**. The new version of this page highlights the advantages of connecting your PMM instance to Percona Platform and makes it easier to generate and use a connection token.  

> - [PMM-11499](https://jira.percona.com/browse/PMM-11499) - You can now enable Access Control from the **Configuration > Settings > Advaned setting** page on the PMM UI.

Reviewer 3: ```suggestion
- [PMM-11499](https://jira.percona.com/browse/PMM-11499) - You can enable Access Control from the **Configuration > Settings > Advanced Setting** page on the PMM UI.
```

> - [PMM-11499](https://jira.percona.com/browse/PMM-11499) - You can now enable Access Control from the **Configuration > Settings > Advaned setting** page on the PMM UI.

Reviewer 3: what do we put this to improvement part? Isnt it the part of Access Control? We're not improving it, we added it on this release. @matejkubinec what do you think?

Reviewer 5: I agree, it's not worth mentioning it under improvements.

Author: Deleted it.

---
name: Percona Monitoring and Management <version>
about: Release Notes Template for PMM

---

| **Release date:** | [Month DD, YYYY]                                                                                    |
| ----------------- | ----------------------------------------------------------------------------------------------- |
| **Installation:** | [Installing Percona Monitoring and Management](https://www.percona.com/software/pmm/quickstart) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

!!! caution alert alert-warning "Important/Caution"
    Crucial points that need emphasis:

    - Important: A significant point that deserves emphasis.
    - Caution: Used to mean 'Continue with care'.

## Release Highlights

> 

Reviewer 6: - PMM-11302 (https://jira.percona.com/browse/PMM-11302) Ability to add backup schedule while creating DB clusters in DBaaS
- PMM-11301 (https://jira.percona.com/browse/PMM-11301) Ability to restore cluster from backup stored on S3 while creating DB clusters in DBaaS

Starting with PMM 2.35.0, we are excited to announce the first step towards PMM Access Management, a highly anticipated solution by our users. With this solution, PMM allows you to manage who has access to individual Prometheus (Victoria Metics)  metrics based on labels. This new feature provides a standardized method for granting, changing, and revoking access to metrics. With the introduction of label-based access control, you can now establish roles that control which data can be queried based on specific label criteria. It allows you to associate multiple sets of individual Prometheus (Victoria Metics) labels with a role, ensuring that queries will only return data from series that meet the defined labels. This provides you with a powerful and flexible way to manage your data and ensure secure access to the information you need. For more information, visit the Access Control documentation.

PMM Access Management will be introduced in phases to ensure a seamless rollout in line with your requirements and feedback. Stay tuned for further updates on our roadmap for PMM Access Control.

> ###  GA: Deployment of PMM in k8 with Helm Chart

Reviewer 2: @dygksy @puneet0191 this is not tested yet

Author: @denisok @puneet0191 should I remove it then?

Reviewer 3: @puneet0191 could we complete the test if possible? Customers are waiting its GA for a long time :/

## Bugs Fixed

> - [PMM-9446](https://jira.percona.com/browse/PMM-9446) - Fixed an issue with the MYSQL/ MYSQL Details dashboard, where the users' names were overlapping.

Reviewer 2: ```suggestion
- [PMM-9446](https://jira.percona.com/browse/PMM-9446) - Fixed an issue with the MySQL/MySQL Details dashboard, where the users' names were overlapping.
```

- [PMM-11369](https://jira.percona.com/browse/PMM-11369) - DBaaS: Dollar sign in operators
- [PMM-11399](https://jira.percona.com/browse/PMM-11399) - Alerting: Empty rule name in pop up messages
- [PMM-11474](https://jira.percona.com/browse/PMM-11474) - DBaaS: Internal server error after migration to 2.34.0

> - [PMM-11517](https://jira.percona.com/browse/PMM-11517 - Restores from physical backups on local storage fails after first attempt

Reviewer 2: ```suggestion
- [PMM-11517](https://jira.percona.com/browse/PMM-11517) - Restores from physical backups on local storage fails after first attempt
```
